presto-main/src/main/java/com/facebook/presto/metadata/OperatorNotFoundException.java

```diff
@@ -40,12 +41,22 @@ public OperatorNotFoundException(OperatorType operatorType, List<? extends Type>
 
     public OperatorNotFoundException(OperatorType operatorType, List<? extends Type> argumentTypes, Type returnType)
     {
-        super(OPERATOR_NOT_FOUND, format("Operator %s(%s):%s not registered", operatorType, Joiner.on(", ").join(argumentTypes), returnType));
+        super(OPERATOR_NOT_FOUND, formatErrorMessage(operatorType, argumentTypes, Optional.of(returnType)));
         this.operatorType = requireNonNull(operatorType, "operatorType is null");
         this.argumentTypes = ImmutableList.copyOf(requireNonNull(argumentTypes, "argumentTypes is null"));
         this.returnType = requireNonNull(returnType, "returnType is null");
     }
 
+    private static String formatErrorMessage(OperatorType operatorType, List<? extends Type> argumentTypes, Optional<Type> returnType)
+    {
+        String returnTypeString = returnType.isPresent() ? ":" + returnType.toString() : "";
+        if (argumentTypes.size() == 2) {
+            return format("'%s'%s not allowed between %s", operatorType.getOperator(), returnTypeString, argumentTypes.get(0) + " and " + argumentTypes.get(1));
+        }
+
+        return format("'%s'%s not allowed for (%s)", operatorType.getOperator(), returnTypeString, Joiner.on(", ").join(argumentTypes));
```

Comment:
This would look strange for `BETWEEN`:

```
'BETWEEN':BOOLEAN not allowed for (BIGINT, VARCHAR, VARCHAR)
```

We probably want a dedicated message for that operator.

Similar comment for CAST.